crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -580,6 +607,58 @@ module sui::validator_set {
         }
     }
 
+    /// Some validators' rewards may get slashed due to getting sub-par scores
+    /// for tallying rule. This function computes, using the report record, the
+    /// stake adjustment amounts of non-performant validators. When computing reward
+    /// distribution, the adjusted stake amount will be used so these validators
+    /// receive less rewards than what's proportional to their original stake.
+    fun compute_stake_adjustments(
+        self: &ValidatorSet,
+        slashed_validators: vector<address>,
+        reward_slashing_rate: u64,
+    ): (u64, VecMap<address, u64>) {
+        let total_adjustment = 0;
+        let individual_adjustments = vec_map::empty();
+        while (!vector::is_empty(&mut slashed_validators)) {
+            let validator_address = vector::pop_back(&mut slashed_validators);
+            let original_stake = validator_total_stake_amount(self, validator_address);
+            let adjustment_u128 = 
+                (original_stake as u128) * (reward_slashing_rate as u128) 
```

Comment:
This confuses me a bit. Why are we multiplying reward_slashing_rate to the validator total stake?